GraphTheory/Dijkstra.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include <vector>
+#include <queue>
+
+#define INF 2147483647
+
+using namespace std;
+
+
+vector<vector< pair<int, int>>> g; // global graph
+int vertices, edges;
+
```

Comment:
- typedef `vector<vector<pair<int, int>>>` to `Graph`,
- Rename `g` to `graph`,
- Make `graph`, `vertices` and `edges` as local variables of `main()`. You can pass the graph as a constant reference (`const Graph& graph`) to `dijkstra()`.